bin/asl-setup-dkms-mok

```diff
@@ -0,0 +1,85 @@
+#!/usr/bin/bash
+#
+
+if [ "${USER}" != "root" ]; then
+	echo "Must be run as root"
+	exit 1
+fi
+
+if [ ! -x /usr/bin/mokutil ]; then
+	echo "Missing /usr/bin/mokutil"
+	exit 1
+fi
+
+SBSTATE=$(mokutil --sb-state | awk '{print $2}')
+if [ "${SBSTATE}" != "enabled" ]; then
+	echo "This system is not running SecureBoot"
+	exit 0
+fi
+
+echo "!!!"
+echo "!!! WARNING - RUNNING THIS SCRIPT WILL REGENERATE THE"
+echo "!!! DEFAULT MACHINE OWNER KEY (MOK) FOR SECURE BOOT. IN GENERAL YOU"
+echo "!!! ARE RUNNING THIS SCRIPT BECAUSE YOU NEED TO AND YOU KNOW WHY"
+echo "!!! YOU ARE DOINGING IT. HOWEVER BE WARNED!"
+echo "!!!"
+read -p "ENTER \"galaxy\" TO CONTINUE>> " proceed
+
+if [ "${proceed}" != "galaxy" ]; then
+	exit 1
+fi
+
+echo ""
+echo "Generating an MOK requires a name for the certificate. This can be"
+echo "anything, however your name or your callsign is recommended."
+echo ""
+read -p "Name or callsign>> " subj
+
+if [ -z "${subj}" ]; then
+	echo "MOK subject cannot be blank"
+	exit 1
+fi
+
+echo ""
+echo "Generating the MOK..."
+test ! -d /var/lib/shim-signed/mok && mkdir -p /var/lib/shim-signed/mok
+pushd /var/lib/shim-signed/mok > /dev/null
+openssl req -nodes -new -x509 -newkey rsa:2048 -keyout MOK.priv \
+	-outform DER -out MOK.der -days 36500 -subj "/CN=${subj}/"
+openssl x509 -inform der -in MOK.der -out MOK.pem
+
+echo ""
+echo "Enter a one-time password for this key. This password will be prompted"
+echo "for at the next reboot to properly install the key in the UEFI"
+echo "bootloader keyring. This does not need to be an incredibly complex"
+echo "password and you may write it down until it's used."
+echo ""
+mokutil --import MOK.der
+popd > /dev/null
+
+if [ $? != 0 ]; then
```

Comment:
I'm not too familiar with the process here, but just want to confirm that this is checking the return value of `popd`?
This command seems very unlikely to fail  unless there is something about `mokutil --import` that is supposed to delete that directory - is this the case?

Reply:
THat's my mistake when I added that on the fly. The $? is supposed to be checking the return of mokutil